with-react-query/search.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react"
+import { useQuery } from "react-query"
+
+type User = {
+  name: string
+  followers: number
+}
+
+const Search = () => {
+  const [username, setUsername] = React.useState(undefined)
+  const { status, error, data } = useQuery<boolean, Error, User>(
+    ["githubUser", username],
+    ({ queryKey }) =>
```

Comment:
Can we use an async callback instead? That'd make it a bit cleaner :D

Reply:
yes